flink-table-store-core/src/main/java/org/apache/flink/table/store/CoreOptions.java

```diff
@@ -474,6 +474,18 @@ public class CoreOptions implements Serializable {
                                                             + "$hour:00:00'."))
                                     .build());
 
+    public static final ConfigOption<String> PRIMARY_KEY =
+            key("primary-key")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Define primary key for `CREATE TABLE AS SELECT` statement.");
```

Comment:
`Define primary key by table options, cannot define primary key on DDL and table options at the same time.`.